src/java.base/share/classes/java/lang/reflect/ProxyGenerator.java

```diff
@@ -952,65 +924,72 @@ public String toString() {
-            add(long.class, Long.class);
-            add(short.class, Short.class);
-            add(boolean.class, Boolean.class);
-        }
+    private enum PrimitiveTypeInfo {
+        BYTE(byte.class, ILOAD),
+        CHAR(char.class, ILOAD),
+        DOUBLE(double.class, DLOAD),
+        FLOAT(float.class, FLOAD),
+        INT(int.class, ILOAD),
+        LONG(long.class, LLOAD),
+        SHORT(short.class, ILOAD),
+        BOOLEAN(boolean.class, ILOAD);
 
         /**
-         * name of corresponding wrapper class
+         * internal name of corresponding wrapper class
          */
-        private String wrapperClassName;
+        private final String wrapperClassName;
         /**
          * method descriptor for wrapper class "valueOf" factory method
          */
-        private String wrapperValueOfDesc;
+        private final String wrapperValueOfDesc;
         /**
          * name of wrapper class method for retrieving primitive value
          */
-        private String unwrapMethodName;
+        private final String unwrapMethodName;
         /**
          * descriptor of same method
          */
-        private String unwrapMethodDesc;
+        private final String unwrapMethodDesc;
+        /**
+         * Load opcode used by this primitive
+         */
+        private final int loadOpcode;
+        /**
+         * Return opcode used by this primitive
+         */
+        private final int returnOpcode;
 
-        private PrimitiveTypeInfo(Class<?> primitiveClass, Class<?> wrapperClass) {
+        PrimitiveTypeInfo(Class<?> primitiveClass, int loadOpcode) {
             assert primitiveClass.isPrimitive();
 
-            /**
-             * "base type" used in various descriptors (see JVMS section 4.3.2)
-             */
-            String baseTypeString =
-                    Array.newInstance(primitiveClass, 0)
-                            .getClass().getName().substring(1);
-            wrapperClassName = dotToSlash(wrapperClass.getName());
+            Wrapper wrapper = Wrapper.forPrimitiveType(primitiveClass);
+            // single-char BaseType descriptor (see JVMS section 4.3.2)
+            String baseTypeString = wrapper.basicTypeString();
+            var wrapperType = wrapper.wrapperType();
+            wrapperClassName = dotToSlash(wrapperType.getName());
             wrapperValueOfDesc =
-                    "(" + baseTypeString + ")L" + wrapperClassName + ";";
+                    "(" + baseTypeString + ")" + wrapperType.descriptorString();
             unwrapMethodName = primitiveClass.getName() + "Value";
             unwrapMethodDesc = "()" + baseTypeString;
-        }
-
-        private static void add(Class<?> primitiveClass, Class<?> wrapperClass) {
-            table.put(primitiveClass,
-                    new PrimitiveTypeInfo(primitiveClass, wrapperClass));
+            this.loadOpcode = loadOpcode;
+            this.returnOpcode = loadOpcode - ILOAD + IRETURN;
```

Comment:
This could do it.  It would be more explicit to take the return opcode as an argument to the constructor.